doc/source/merging.rst

```diff
@@ -714,6 +714,31 @@ either the left or right tables, the values in the joined table will be
           labels=['left', 'right'], vertical=False);
    plt.close('all');
 
+To join a Series and a DataFrame, the Series has to be transformed into a DataFrame first:
```

Comment:
Explain that the `Series` is a MultiIndexed series, and that's what you're joining on. Something like

    To join a Series with a MultiIndex and a DataFrame, using the levels of the
    MultiIndex and columns from the DataFrame, transform the Series to a DataFrame
    using :meth:`Series.reset_index`.